sqlitecluster/SQLiteNode.cpp

```diff
@@ -1870,18 +1917,24 @@ void SQLiteNode::_sendToPeer(SQLitePeer* peer, const SData& message) {
     // We can treat this whole function as atomic and thread-safe as it sends data to a peer with it's own atomic
     // `sendMessage` and the peer itself (assuming it's something from _peerList, which, if not, don't do that) is
     // const and will exist without changing until destruction.
-    peer->sendMessage(_addPeerHeaders(message).serialize());
+    if (peer->forked) {
+        PINFO("Skipping message " << message.methodLine << " to forked peer.");
+    }
+    peer->sendMessage(_addPeerHeaders(message));
 }
 
 void SQLiteNode::_sendToAllPeers(const SData& message, bool subscribedOnly) {
-    const string serializedMessage = _addPeerHeaders(message).serialize();
+    const SData messageWithHeaders = _addPeerHeaders(message);
 
     // Loop across all connected peers and send the message. _peerList is const so this is thread-safe.
     for (auto peer : _peerList) {
+        if (peer->forked) {
+            PINFO("Skipping message " << message.methodLine << " to forked peer.");
```

Comment:
I might be misunderstanding these log lines, if so, ignore my suggestions
```suggestion
            PINFO("Skipping message " << message.methodLine << " to forked peer.");
            continue;
```